source/language/standard_library.rst

```diff
@@ -0,0 +1,329 @@
+Standard library
+================
+
+The standard library for OpenQASM 3 consists of a single include file, called ``stdgates.inc``.
+This file can be included in an OpenQASM 3 program with the statement
+
+.. code-block::
+
+   include "stdgates.inc";
+
+Versioning
+----------
+
+The standard library is versioned directly with the language itself.  This documentation will note
+the version that any given gate in the standard library became available, and, if applicable, the
+version it was removed in.
+
+Notes for implementations
+-------------------------
+
+An implementation is not required to make the standard library available in situations where the
+OpenQASM 3 program is being interpreted in terms of the instruction set architecture of a particular
+QPU.  The rest of this section concerns only situations where the library is considered available.
+
+Implementations should consider the ``stdgates.inc`` file to be locatable, regardless of any other
+settings that they might apply around include search paths.
+
+Upon interpreting an ``include "stdgates.inc";`` statement, implementations should make available
+exactly the set of gates defined in the standard library for the language version that matches the
+:ref:`version-string statement <version-string>`, if present.  If the version statement is not
+present, the implementation should make available the gates from the language version it is parsing
+the rest of the program under.
+
+Implementations must not define gates that are not present in chosen OpenQASM 3 version, where
+doing so could cause a user program to be ill formed (for example due to a user attempting to define
+their own gate with a clashing name).
+
+An implementation is not required to supply a literal ``stdgates.inc`` file to the user.
```

Comment:
Should we update https://openqasm.com/language/comments.html#included-files in any way?

Reply:
What kind of thing were you thinking?  We should probably put in a link to this page as a cross-reference to the `stdlib.inc` file for certain.

Reply:
I've included some cross-referencing to other places in 885ab15 - happy to add more.

Reply:
Looks good, thank you.